packages/utils/src/post-process.ts

```diff
@@ -9,8 +9,48 @@ import {
 } from '@babel/types';
 import { transformSync, Node, Visitor, template, PluginObj } from '@babel/core';
 
+/**
+ * Source map declaration taken from `@babel/core`. Babel doesn't export the
+ * type for this, so it's copied from the source code instead here.
+ */
+export type SourceMap = {
+  version: number;
+  sources: string[];
+  names: string[];
+  sourceRoot?: string | undefined;
+  sourcesContent?: string[] | undefined;
+  mappings: string;
+  file: string;
+};
+
+/**
+ * The post process options.
+ *
+ * @property stripComments - Whether to strip comments. Defaults to `true`.
+ * @property sourceMap - Whether to generate a source map for the modified code.
+ * See also `inputSourceMap`.
+ * @property inputSourceMap - The source map for the input code. When provided,
+ * the source map will be used to generate a source map for the modified code.
+ * This ensures that the source map is correct for the modified code, and still
+ * points to the original source. If not provided, a new source map will be
+ * generated instead.
+ */
 export type PostProcessOptions = {
-  stripComments: boolean;
+  stripComments?: boolean;
+  sourceMap?: boolean | 'inline';
+  inputSourceMap?: SourceMap;
+};
+
+/**
+ * The post processed bundle output.
+ *
+ * @property code - The modified code.
+ * @property sourceMap - The source map for the modified code, if the source map
+ * option was enabled.
+ */
+export type PostProcessedBundle = {
+  code: string;
+  sourceMap?: SourceMap | null;
```

Comment:
this can have 3 different options because of `?`. `SourceMap`, `null` and `undefined`. Is it really what you intended?

Reply:
Yeah, because the `file.map` property from Babel has the same signature, for some reason.